cjdns-hdr/src/lib.rs

```diff
@@ -0,0 +1,10 @@
+mod data_header;
+mod errors;
+mod route_header;
+mod switch_header;
+mod utils;
+
+// TODO
+// 1. Tests for utils reader and writer
```

Comment:
[Actually, this should be one of the first things to do, not the last. It is always better and easier to develop having some tests (aka TDD). Also when one start to write higher-level logic (parsers), it is better to have some confidence that your building blocks (Reader/Writer) are working properly, so you don't need to go back and debug them while working on other part of the program.]